File: oj_server/questions/5/java/tail.java
// ======== TAIL ========
public class Main {

    @FunctionalInterface
    interface TestFunc {
        boolean run();
    }

    static class TestCase {
        TestFunc testFunc;
        String inputStr;

        TestCase(TestFunc testFunc, String inputStr) {
            this.testFunc = testFunc;
            this.inputStr = inputStr;
        }
    }

    static boolean assertEqual(boolean actual, boolean expected) {
        return actual == expected;
    }

    static int saveResult(List<TestCase> tests, String resultPath) {
        int totalTests = tests.size();
        int passedTests = 0;
        String firstFailedInput = "";

        for (TestCase test : tests) {
            if (test.testFunc.run()) {  // 直接调用 run 方法
                passedTests++;
            } else if (firstFailedInput.isEmpty()) {
                firstFailedInput = test.inputStr;
            }
        }

        try (FileWriter writer = new FileWriter(resultPath)) {
            writer.write(passedTests + " " + totalTests + " " + firstFailedInput);
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
            return 1;
        }

        return (passedTests == totalTests) ? 0 : 1;
    }

    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Usage: java Main <result_file_path>");
            System.exit(1);
        }

        String resultPath = args[0];

        List<TestCase> tests = new ArrayList<>();
        tests.add(new TestCase(
                () -> assertEqual(new Solution().isPalindrome(121), true),
                "121"
        ));
        tests.add(new TestCase(
                () -> assertEqual(new Solution().isPalindrome(-10), false),
                "-10"
        ));
        tests.add(new TestCase(
                () -> assertEqual(new Solution().isPalindrome(12321), true),
                "12321"
        ));

        int result = saveResult(tests, resultPath);
        System.exit(result);
    }
}

File: oj_server/questions/7/go/tail.go

func assertEqual(actual, expected bool) bool {
	return actual == expected
}

type TestCase struct {
	testFunc func() bool
	inputStr string
}

func saveResult(tests []TestCase, resultPath string) int {
	totalTests := len(tests)
	passedTests := 0
	firstFailedInput := ""

	for _, test := range tests {
		if test.testFunc() {
			passedTests++
		} else if firstFailedInput == "" {
			firstFailedInput = test.inputStr
		}
	}

	file, err := os.Create(resultPath)
	if err != nil {
		fmt.Printf("Error creating file: %v\n", err)
		return 1
	}
	defer file.Close()

	result := fmt.Sprintf("%d %d %s", passedTests, totalTests, firstFailedInput)
	_, err = file.WriteString(result)
	if err != nil {
		fmt.Printf("Error writing to file: %v\n", err)
		return 1
	}

	if passedTests == totalTests {
		return 0
	}
	return 1
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: program <result_file_path>")
		os.Exit(1)
	}

	resultPath := os.Args[1]

	tests := []TestCase{
		{
			testFunc: func() bool {
				return assertEqual(Solution{}.isPalindrome(121), true)
			},
			inputStr: "121",
		},
		{
			testFunc: func() bool {
				return assertEqual(Solution{}.isPalindrome(-10), false)
			},
			inputStr: "-10",
		},
		{
			testFunc: func() bool {
				return assertEqual(Solution{}.isPalindrome(12321), true)
			},
			inputStr: "12321",
		},
	}

	os.Exit(saveResult(tests, resultPath))
}
